libgpos/src/_api.cpp

```diff
@@ -44,9 +44,17 @@ bool fEnableExtendedAsserts = false;  // by default, extended asserts are disabl
 //		Initialize GPOS memory pool, worker pool and message repository
 //
 //---------------------------------------------------------------------------
-void __attribute__((constructor)) gpos_init()
-{
-	if (GPOS_OK != gpos::CMemoryPoolManager::EresInit())
+void gpos_init(struct gpos_init_params* params) {
```

Comment:
Had an offline discussion. I don't think performance matters much here (since the difference is very small and only happens once at init). I'm following the same conventions as gpos_exec_params by passing it by reference. These functions are most likely to be static functions compiled with GPDB, so I don't expect a leak if passed on the stack.